src/DynamoCore/Search/SearchElementGroup.cs

```diff
@@ -4,4 +4,13 @@ public enum SearchElementGroup
     {
         None, Create, Action, Query
     }
+
+    public enum ElementTypeEnum
+    {
+        RegularNode,
+        RegularCategory,
+        CustomNode,
+        CustomDll,
+        Package
+    }
```

Comment:
Can we change this to the following flags:

```csharp
public enum ElementTypes
{
    None        = 0x00000000,
    ZeroTouch   = 0x00000001, // All the DLLs (can be built-in, third-party, and packaged)
    CustomNode  = 0x00000002, // All the DYFs (both loose DYF files and packaged)
    BuiltIn     = 0x00010000, // Things that Dynamo ships with (operator, geometries...)
    Packaged    = 0x00020000, // Packaged element (either zero-touch DLLs or DYFs)
}
```

1. `None` above is the same as your original `RegularCategory`.

2. We can only have DLL or DYF file types.
3. DLLs can be either of the following types:

 - built-in (e.g. geometries): type = ZeroTouch | BuiltIn
 - loose DLL (e.g. ones user built): type = ZeroTouch
 - packaged (e.g. downloaded from package manager): type = ZeroTouch | Packaged

4. DYFs can be either of the following types:

 - loose DYF files (e.g. ones user built): type = CustomNode
 - packaged (e.g. downloaded from package manager): type = CustomNode | Packaged